osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/BatchInserter.cs

```diff
@@ -84,44 +83,39 @@ private async Task run(HighScore[] scores)
                 int rulesetId = ruleset.RulesetInfo.OnlineID;
 
                 // check for existing and skip
-                SoloScoreLegacyIDMap[] existingIds = (await db.QueryAsync<SoloScoreLegacyIDMap>(
-                    $"SELECT * FROM score_legacy_id_map WHERE `ruleset_id` = {rulesetId} AND `old_score_id` IN @oldScoreIds",
+                SoloScore[] existingIds = (await db.QueryAsync<SoloScore>(
+                    $"SELECT id, legacy_score_id FROM scores WHERE `ruleset_id` = {rulesetId} AND `legacy_score_id` IN @legacyScoreIds",
                     new
                     {
-                        oldScoreIds = scores.Select(s => s.score_id)
+                        legacyScoreIds = scores.Select(s => s.score_id)
                     }, transaction)).ToArray();
 
                 insertCommand.CommandText =
                     // main score insert
-                    "INSERT INTO scores (user_id, beatmap_id, ruleset_id, data, has_replay, preserve, created_at, unix_updated_at) "
-                    + $"VALUES (@userId, @beatmapId, {rulesetId}, @data, @has_replay, 1, @date, UNIX_TIMESTAMP(@date));";
+                    "INSERT INTO scores (`user_id`, `ruleset_id`, `beatmap_id`, `has_replay`, `preserve`, `rank`, `passed`, `accuracy`, `max_combo`, `total_score`, `data`, `pp`, `legacy_score_id`, `legacy_total_score`, `ended_at`, `unix_updated_at`) "
+                    + $"VALUES (@user_id, {rulesetId}, @beatmap_id, @has_replay, 1, @rank, 1, @accuracy, @max_combo, @total_score, @data, @pp, @legacy_score_id, @legacy_total_score, @ended_at, UNIX_TIMESTAMP(@ended_at));";
 
-                // pp insert
-                if (importLegacyPP)
-                    insertCommand.CommandText += "INSERT INTO score_performance (score_id, pp) VALUES (LAST_INSERT_ID(), @pp);";
+                updateCommand.CommandText = "UPDATE scores SET data = @data WHERE id = @id";
```

Comment:
Isn't this update very wrong now? It used to update stuff that is now hoisted out of the json blob into columns. `pp` is probably of most immediate concern.

Reply:
yep, probably. wasn't paying much mind to update since we don't use that atm.

Reply:
Let's overlook this for now. I'm removing the update/delete flow from `BatchInserter` (will use `Verify` command for that in the future).